woocommerce/woocommerce-paypal-payments
# Add quick setup methods for recommended payment gateways

Adds quick setup methods for the payment recommendations setup.

Related PR - https://github.com/woocommerce/woocommerce-admin/pull/7044

---

### Description

This PR SlotFills the payment task in the WooCommerce onboarding task list.  In brief, this PR does the following:

* Adds a webpack build process to the `ppcp-onboarding` module.
* Adds a script to SlotFill the payment task in WooCommerce onboarding.
* Updates the gateway's `needs_setup` check.
* Register (but not enqueue) scripts so we can expose and enqueue them dynamically via the REST API.

I apologize for the size of this PR; most if it is boilerplate and the component itself has been copied from [WooCommerce Admin](https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks/payments/methods/paypal.js) with a bit of refactoring.

I tried to follow this repo's conventions as closely as possible, but please let me know if there are changes that would be more in line with the code in this repo.

### Steps to test:

1. Install and activate this plugin in your `plugins/` directory matching the wp.org slug.  `woocommerce-paypal-payments`
2. Build out the modules `npm run build:dev`;
2. Install and activate the development version of [WooCommerce Admin](https://github.com/woocommerce/woocommerce-admin) ( `git clone https://github.com/woocommerce/woocommerce-admin.git && cd woocommerce-admin && composer install && npm i && npm run build` ).
3. Make sure India is not selected as the store country under WooCommerce general settings and CBD is not selected as an industry during onboarding.
2. Navigate to the PayPal gateway in the payments task `/wp-admin/admin.php?page=wc-admin&task=payments`
3. Check that the "Connect" button works and opens the PayPal connection flow. (Note the return to WooCommerce Admin still needs tweaking).
4. Throw an error in the `/onboarding/get-params` endpoint to force the manual connection form.
5. Fill out the manual connection form and update settings.
4. Make sure that the PayPal gateway is moved to the "Enabled payment gateways" section.
5. Make sure that settings are persisted on when visiting the Payment settings under WooCommerce->Settings. (I'm getting a bearer token error on the settings page when setting bad credentials- is this an error in setting or should further sanitization happen in the controller?)

### Changelog entry
> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Add SlotFill for WooCommerce Admin onboarding payment task.
